### ISSUE TITLE
Recursion

### ISSUE THREAD

**Issue by bendc** [babel/babel#1589]
The following code:
```javascript
const count = (i=10) => {
  if (!i) return;
  console.log(i);
  return count(i-1);
};
```
Generates the following output, which crashes all browsers I've tested:
```javascript
var count = function count() {
  var _again = true;

  _function: while (_again) {
    i = undefined;
    _again = false;
    var i = arguments[0] === undefined ? 10 : arguments[0];

    if (!i) return;
    console.log(i);
    i - 1;
    _again = true;
    continue _function;
  }
};
```
I've successfully used recursion with Babel in the past, is that a regression?

**Comment by sebmck** [comment]
It's the TCO. Temporary fix is to blacklist `es6.tailCall`. It doesn't seem to be working nicely with default parameters.

**Comment by sebmck** [comment]
Fixed as of 5.4.5, thanks!
